core/src/main/java/nl/nn/adapterframework/webcontrol/api/Base.java

```diff
@@ -199,4 +203,17 @@ protected String resolveStringFromMap(Map<String, List<InputPart>> inputDataMap,
 		}
 		throw new ApiException("Key ["+key+"] not defined", 400);
 	}
+	
+	private Optional<String> getAttributeValue(String name, List<Attachment> attachmentList) {
+		return attachmentList.stream().filter(att -> name.equalsIgnoreCase(att.getDataHandler().getDataSource().getName())).findAny().map(m -> {
```

Comment:
This has become pretty unreadable. Try to avoid using lambda's unless they are short and simple.

Reply:
I have removed some lambda expression, now it is more readable. let me know what do you think about the new code please